src/compiler/scala/tools/nsc/util/ClassPath.scala

```diff
@@ -128,7 +128,10 @@ object ClassPath {
   def split(path: String): List[String] = (path split pathSeparator).toList.filterNot(_ == "").distinct
 
   /** Join classpath using platform-dependent path separator */
-  def join(paths: String*): String  = paths filterNot (_ == "") mkString pathSeparator
+  def join(paths: String*): String  = paths.toList.filterNot(_ == "") match {
+    case only :: Nil => only
```

Comment:
Would not this case be covered by the second one? Or does this provide some optimisation?

Reply:
`mkString` is not optimized for the special case of a one-element collection containing a `String`. That would be worthwhile as a separate change on 2.13.x.